core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java

```diff
@@ -0,0 +1,314 @@
+        RunContext runContext,
+        WorkerTaskExecution<?> workerTaskExecution,
+        Flow flow,
+        Execution execution
+    ) {
+        TaskRun taskRun = workerTaskExecution.getTaskRun();
+
+        taskRun = taskRun.withState(ExecutableUtils.guessState(execution, this.subflow.transmitFailed));
+
+        int currentIteration = (Integer) taskRun.getOutputs().get("currentIteration");
+        int maxIterations = (Integer) taskRun.getOutputs().get("maxIterations");
+
+        return currentIteration == maxIterations ? Optional.of(ExecutableUtils.workerTaskResult(taskRun)) : Optional.empty();
+    }
+
+    @Override
+    public boolean waitForExecution() {
+        return this.subflow.wait;
+    }
+
+    private URI readItems(Execution execution, String taskRunId, int split) {
+        // Recreate the URI from the execution context and the value.
+        // It should be kestra:///$ns/$flow/executions/$execution_id/tasks/$task_id/$taskrun_id/bach-$value.ion
+        String uri = URI_FORMAT.formatted(execution.getNamespace(), execution.getFlowId(), execution.getId(), this.id, taskRunId, split);
+        return URI.create(uri);
+    }
+
+    private int readSplits(RunContext runContext) throws IllegalVariableEvaluationException {
+        URI data = URI.create(runContext.render(this.items));
+
+        try (var reader = new BufferedReader(new InputStreamReader(runContext.uriToInputStream(data)))) {
+            int batches = 0;
+            int lineNb = 0;
+            String row;
+            List<String> rows = new ArrayList<>(maxItemsPerBatch);
+            while ((row = reader.readLine()) != null) {
+                rows.add(row);
+                lineNb++;
+
+                if (lineNb == maxItemsPerBatch) {
+                    createBatchFile(runContext, rows, batches);
+
+                    batches++;
+                    lineNb = 0;
+                    rows.clear();
+                }
+            }
+
+            if (!rows.isEmpty()) {
+                createBatchFile(runContext, rows, batches);
+                batches++;
+            }
+
+            return batches;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void createBatchFile(RunContext runContext, List<String> rows, int batch) throws IOException {
+        byte[] bytes = rows.stream().collect(Collectors.joining(System.lineSeparator())).getBytes();
+        File batchFile = runContext.tempFile(bytes, ".ion").toFile();
+        runContext.putTempFile(batchFile, "bach-" + batch + ".ion");
+    }
+
+    @SuperBuilder
+    @ToString
+    @EqualsAndHashCode
+    @Getter
+    @NoArgsConstructor
+    public static class Subflow {
+        @NotEmpty
+        @Schema(
+            title = "The namespace of the flow to trigger"
+        )
+        @PluginProperty(dynamic = true)
+        private String namespace;
+
+        @NotEmpty
+        @Schema(
+            title = "The identifier of the flow to trigger"
```

Comment:
```suggestion
            title = "The identifier of the flow to be executed"
```
you didn't say it wrong, only thinking that we may say `execute` instead of `trigger` to link it to the concept of Execution rather than the concept of a Trigger